lib/ddtrace/contrib/pg/connection.rb

```diff
@@ -0,0 +1,88 @@
+
+module Datadog
+  module Contrib
+    module Pg
+      # PG::Connection patch module
+      module Connection
+        module_function
+
+        def included(base)
+          base.send(:prepend, InstanceMethods)
+        end
+
+        # PG::Connection patch instance methods
+        module InstanceMethods
+
+          # sync_exec(sql) -> PG::Result
+          # sync_exec(sql) {|pg_result| block}
+          def sync_exec(sql)
+            datadog_pin.tracer.trace(Ext::SPAN_QUERY) do |span|
+              span.resource = sql
+              span.service = datadog_pin.service
+              span.span_type = Datadog::Ext::SQL::TYPE
+              span.set_tag(Ext::TAG_DB_NAME, db)
+              span.set_tag(Datadog::Ext::NET::TARGET_HOST, host)
+              span.set_tag(Datadog::Ext::NET::TARGET_PORT, port)
+              super # this will pass all args, including the block
+            end
+          end
+
+          # sync_exec_params(sql, params[, result_format[, type_map]] ) -> PG::Result
+          # sync_exec_params(sql, params[, result_format[, type_map]] ) {|pg_result| block }
+          def sync_exec_params(sql)
+            datadog_pin.tracer.trace(Ext::SPAN_QUERY) do |span|
+              span.resource = sql
+              span.service = datadog_pin.service
+              span.span_type = Datadog::Ext::SQL::TYPE
+              span.set_tag(Ext::TAG_DB_NAME, db)
+              span.set_tag(Datadog::Ext::NET::TARGET_HOST, host)
+              span.set_tag(Datadog::Ext::NET::TARGET_PORT, port)
+              super # this will pass all args, including the block
+            end
+          end
+
+          # async_exec(sql) -> PG::Result OR
+          # async_exec(sql) {|pg_result| block}
+          def async_exec(sql)
+            datadog_pin.tracer.trace(Ext::SPAN_QUERY) do |span|
+              span.resource = sql
+              span.service = datadog_pin.service
+              span.span_type = Datadog::Ext::SQL::TYPE
+              span.set_tag(Ext::TAG_DB_NAME, db)
+              span.set_tag(Datadog::Ext::NET::TARGET_HOST, host)
+              span.set_tag(Datadog::Ext::NET::TARGET_PORT, port)
+              super # this will pass all args, including the block
+            end
+          end
+
+          # async_exec_params(sql, params[, result_format[, type_map]] ) -> PG::Result
+          # async_exec_params(sql, params[, result_format[, type_map]] ) {|pg_result| block }
+          def async_exec_params(sql)
+            datadog_pin.tracer.trace(Ext::SPAN_QUERY) do |span|
+              span.resource = sql
+              span.service = datadog_pin.service
+              span.span_type = Datadog::Ext::SQL::TYPE
+              span.set_tag(Ext::TAG_DB_NAME, db)
+              span.set_tag(Datadog::Ext::NET::TARGET_HOST, host)
+              span.set_tag(Datadog::Ext::NET::TARGET_PORT, port)
+              super # this will pass all args, including the block
+            end
+          end
+
+          def datadog_pin
```

Comment:
`Datadog::Pin` is typically used when you want a specific client/object to have specific settings over the ones configured. This might be appropriate here, if someone would reasonably call `Datadog.configure(connection, service_name: 'my-db')` to do this.

One downside here is if the connection is initialized and the pin accessed, then configuration from the integration will no longer be inherited. Fine for most cases I think, but if a `pg` connection is created and run sometime before Datadog is configured (presumably during the application startup phase, as is sometimes seen with ActiveRecord), then you might not see the expected settings.

Might be okay as-is, just offering this as a consideration.